lib/beaker/cli.rb

```diff
@@ -187,13 +220,35 @@ def print_env_vars_affecting_beaker( log_level )
     # Prints the command line that can be called to reproduce this run
     # (assuming the environment is the same)
     # @param [Symbol] log_level The log level (coloring) to print the message at
-    # @example Print pertinent env vars using error leve reporting (red)
+    # @example Print pertinent env vars using error level reporting (red)
     #     print_command_line :error
     #
+    # @note Re-use of already provisioned SUTs has been tested against the vmpooler & vagrant boxes.
+    #     Fusion doesn't need this, as it has no cleanup steps. Docker is untested at this time.
+    #     Please contact @electrical or the Puppet QE Team for more info, or for requests to support this.
+    #
     # @return nil
     def print_command_line( log_level = :debug )
       @logger.send(log_level, "\nYou can reproduce this run with:\n")
       @logger.send(log_level, @options[:command_line])
+      if @options[:hosts_preserved_yaml_file]
+        set_docker_warning = false
+        has_supported_hypervisor = false
+        @hosts.each do |host|
+          case host[:hypervisor]
+          when /vagrant|fusion|vmpooler|vcloud/
+            has_supported_hypervisor = true
+          when /docker/
+            set_docker_warning = true
+          end
+        end
+        if has_supported_hypervisor
+          @logger.send(log_level, "\nYou can re-run commands against the already provisioned SUT(s) by following these steps:\n")
+          @logger.send(log_level, '(docker support is untested for this feature. please reference the docs for more info)') if set_docker_warning
+          @logger.send(log_level, "- change the hosts file to #{@options[:hosts_preserved_yaml_file]}")
```

Comment:
I would prefer that it would just give you the whole command line to execute instead of having the user have to correctly copy/paste bits and pieces and add the --no-provision.